tracer/src/Datadog.Trace/AppSec/ControllerContextExtensions.Framework.cs

```diff
@@ -28,43 +28,55 @@ internal static void MonitorBodyAndPathParams(this IControllerContext controller
+                if (security.Enabled || iastEnabled)
                 {
-                    if (controllerContext.RouteData.Values.ContainsKey(item.Key))
-                    {
-                        pathParamsDic[item.Key] = item.Value;
-                    }
-                    else
+                    scope = SharedItems.TryPeekScope(context, peekScopeKey);
+
+                    var bodyDic = new Dictionary<string, object>(parameters.Count);
+                    pathParamsDic = new Dictionary<string, object>(parameters.Count);
+                    foreach (var item in parameters)
                     {
-                        bodyDic[item.Key] = item.Value;
+                        if (controllerContext.RouteData.Values.ContainsKey(item.Key))
+                        {
+                            pathParamsDic[item.Key] = item.Value;
+                        }
+                        else
+                        {
+                            // We exclude the query string params
+                            if (!context.Request.QueryString.AllKeys.Contains(item.Key))
+                            {
+                                bodyDic[item.Key] = item.Value;
+                            }
+                        }
                     }
+
+                    requestBody = ObjectExtractor.Extract(bodyDic);
                 }
 
-                var securityTransport = new Coordinator.SecurityCoordinator(security, context, scope.Span);
-                if (!securityTransport.IsBlocked)
+                if (security.Enabled)
                 {
-                    var inputData = new Dictionary<string, object>
+                    var securityTransport = new Coordinator.SecurityCoordinator(security, context, scope.Span);
+                    if (!securityTransport.IsBlocked)
                     {
-                        { AddressesConstants.RequestBody, ObjectExtractor.Extract(bodyDic) },
-                        { AddressesConstants.RequestPathParams, ObjectExtractor.Extract(pathParamsDic) }
-                    };
-                    securityTransport.CheckAndBlock(inputData);
+                        var inputData = new Dictionary<string, object>
+                        {
+                            { AddressesConstants.RequestBody, requestBody },
+                            { AddressesConstants.RequestPathParams, ObjectExtractor.Extract(pathParamsDic) }
+                        };
+                        securityTransport.CheckAndBlock(inputData);
+                    }
                 }
-            }
 
-            if (iastEnabled)
-            {
-                scope?.Span?.Context?.TraceContext?.IastRequestContext?.AddRequestData(context.Request, controllerContext.RouteData.Values);
+                if (iastEnabled)
+                {
+                    scope?.Span?.Context?.TraceContext?.IastRequestContext?.AddRequestData(context.Request, controllerContext.RouteData.Values);
```

Comment:
It might be better to store `scope?.Span?.Context?.TraceContext?.IastRequestContext` to a local variable once and reuse it as the generated low level c # appears to be: 
```
 if (iastEnabled)
      {
        if (scope != null)
        {
          Span span = scope.Span;
          if (span != null)
          {
            SpanContext context2 = span.Context;
            if (context2 != null)
            {
              TraceContext traceContext = context2.TraceContext;
              if (traceContext != null)
              {
                IastRequestContext iastRequestContext = traceContext.IastRequestContext;
                if (iastRequestContext != null)
                  iastRequestContext.AddRequestData(context1.Request, controllerContext.RouteData.Values);
              }
            }
          }
        }
        if (scope != null)
        {
          Span span = scope.Span;
          if (span != null)
          {
            SpanContext context3 = span.Context;
            if (context3 != null)
            {
              TraceContext traceContext = context3.TraceContext;
              if (traceContext != null)
              {
                IastRequestContext iastRequestContext = traceContext.IastRequestContext;
                if (iastRequestContext != null)
                  iastRequestContext.AddRequestBody((object) null, requestBody);
              }
            }
          }
        }
      }
```

Reply:
Good catch! Thank you!